vlog4j-examples/src/main/java/org/semanticweb/vlog4j/examples/core/AddDataFromRdfFile.java

```diff
@@ -0,0 +1,167 @@
+import org.semanticweb.vlog4j.core.reasoner.exceptions.IncompatiblePredicateArityException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.ReasonerStateException;
+import org.semanticweb.vlog4j.core.reasoner.implementation.RdfFileDataSource;
+import org.semanticweb.vlog4j.examples.ExamplesUtils;
+
+/**
+ * This example is an adaptation of {@link AddDataFromCsvFile}. It shows how
+ * facts can be imported from {@code .nt.gz} files, and also how query answers
+ * can be exported to {@code .csv} files. <i>Note that you can also import from
+ * {@code .nt} files.</i>
+ *
+ * For importing, an {@link RdfFileDataSource} that contains a path to the
+ * corresponding {@code .nt.gz} file must be created. An {@code .nt} file
+ * contains facts over one or more predicates in the N-Triples format, while an
+ * {@code .nt.gz} file is the gzipped version of such an {@code .nt} file. For
+ * exporting, a path to the output {@code .csv} file must be specified.
+ *
+ * @author Christian Lewe
+ *
+ */
+public class AddDataFromRdfFile {
+
+	public static void main(final String[] args)
+			throws EdbIdbSeparationException, IOException, ReasonerStateException, IncompatiblePredicateArityException {
+
+		/* 1. Instantiating entities and rules. */
+		final Predicate triplesEDB = makePredicate("triplesEDB", 3);
+		final Predicate triplesIDB = makePredicate("triplesIDB", 3);
+
+		final Constant hasPartPredicate = makeConstant("<http://an.example/hasPart>");
+		final Constant isPartOfPredicate = makeConstant("<http://an.example/isPartOf>");
+		final Constant hasTypePredicate = makeConstant("<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>");
+		final Constant bicycleObject = makeConstant("<http://an.example/bicycle>");
+		final Constant wheelObject = makeConstant("<http://an.example/wheel>");
+
+		final Variable x = makeVariable("x");
+		final Variable s = makeVariable("s");
+		final Variable o = makeVariable("o");
+
+		/*
+		 * We will write '<~/someName>' instead of <http://an.example/someName> and
+		 * <~#someName> instead of <www.w3.org/1999/02/22-rdf-syntax-ns#someName>.
+		 *
+		 * triplesIDB(?s, <~#type>, ?o) :- triplesEDB(?s, <~#type>, ?o) .
+		 */
+		final Atom hasTypeIDB = makeAtom(triplesIDB, s, hasTypePredicate, o);
+		final Atom hasTypeEDB = makeAtom(triplesEDB, s, hasTypePredicate, o);
+		final Rule rule1 = makeRule(hasTypeIDB, hasTypeEDB);
+
+		/*
+		 * triplesIDB(?s, <~/hasPart>, ?o) :- triplesEDB(?s, <~/hasPart>, ?o) .
+		 */
+		final Atom hasPartIDB = makeAtom(triplesIDB, s, hasPartPredicate, o);
+		final Atom hasPartEDB = makeAtom(triplesEDB, s, hasPartPredicate, o);
+		final Rule rule2 = makeRule(hasPartIDB, hasPartEDB);
+
+		/*
+		 * triplesIDB(?s, <~/isPartOf>, ?o) :- triplesEDB(?s, <~/isPartOf>, ?o) .
+		 */
+		final Atom isPartOfIDB = makeAtom(triplesIDB, s, isPartOfPredicate, o);
+		final Atom isPartOfEDB = makeAtom(triplesEDB, s, isPartOfPredicate, o);
+		final Rule rule3 = makeRule(isPartOfIDB, isPartOfEDB);
```

Comment:
As the middle term of these ternary predicates is seen by the reasoner as a simple term (it does not have the interpretation of a predicate), you can have a single rule instead of rule1, rule2 and rule3:

final Variable p = makeVariable("p");
	/*
		 * We will write '<~/someName>' instead of <http://an.example/someName> and
		 * <~#someName> instead of <www.w3.org/1999/02/22-rdf-syntax-ns#someName>.
		 *
		 * triplesIDB(?s,  ?p, ?o) :- triplesEDB(?s, ?p, ?o) .
		 */
		final Atom atomIDB = makeAtom(triplesIDB, s, p, o);
		final Atom atomEDB = makeAtom(triplesEDB, s, p, o);
		final Rule rule1 = makeRule(atomIDB , atomEDB );